program/src/duplicate_block_proof.rs

```diff
@@ -144,17 +113,62 @@ impl<'a> SlashingProofData<'a> for DuplicateBlockProofData<'a> {
         if shred2.len() < shred2_length {
             return Err(SlashingError::ProofBufferTooSmall);
         }
+        let (shred2, _) = shred2.split_at(shred2_length);
+
+        Ok(Self { shred1, shred2 })
+    }
 
+    /// Given the maximum size of a shred as `shred_size` this returns
+    /// the maximum size of the account needed to store a
+    /// `DuplicateBlockProofData`
+    pub const fn size_of(shred_size: usize) -> usize {
+        2usize
+            .wrapping_mul(shred_size)
+            .saturating_add(2 * Self::LENGTH_SIZE)
+    }
+}
+
+impl<'a> SlashingProofData<'a> for DuplicateBlockProofData<'a> {
+    const PROOF_TYPE: ProofType = ProofType::DuplicateBlockProof;
+    type Context = DuplicateBlockProofContext<'a>;
+
+    /// Gives the size of the current proof
```

Comment:
This is used to determine how big the report pda account should be, the `size_of` function above can be used to create and populate the proof account with the record program

Reply:
I like `packed_len`, made the separation here 1115997ce6a48450430a6ac341b7692a512cd5ce``